libminifi/src/core/ProcessSession.cpp

```diff
@@ -1034,14 +1030,18 @@ void ProcessSession::ensureNonNullResourceClaim(
 }
 
 std::shared_ptr<core::FlowFile> ProcessSession::get() {
-  std::shared_ptr<Connectable> first = process_context_->getProcessorNode()->pickIncomingConnection();
+  const auto first = process_context_->getProcessorNode()->pickIncomingConnection();
 
   if (first == nullptr) {
     logger_->log_trace("Get is null for %s", process_context_->getProcessorNode()->getName());
     return nullptr;
   }
 
-  std::shared_ptr<Connection> current = std::static_pointer_cast<Connection>(first);
+  auto current = dynamic_cast<Connection*>(first);
+  if (!current) {
+    logger_->log_error("Get could not find current Connection for %s", process_context_->getProcessorNode()->getName());
```

Comment:
I suggest a more detailed error message:
```suggestion
    logger_->log_error("The incoming connection [%s] of the processor [%s] \"%s\" is not actually a Connection.",
        first->getUUIDStr(), process_context_->getProcessorNode()->getUUIDStr(), process_context_->getProcessorNode()->getName());
```

Reply:
Done.